src/Core/Components/DataGrid/FluentDataGrid.razor

```diff
@@ -0,0 +1,264 @@
+﻿@using Microsoft.AspNetCore.Components.Rendering
+@using Microsoft.FluentUI.AspNetCore.Components.DataGrid.Infrastructure
+@using System.Globalization
+@namespace Microsoft.FluentUI.AspNetCore.Components
+@inherits FluentComponentBase
+@typeparam TGridItem
+<CascadingValue TValue="InternalGridContext<TGridItem>" IsFixed="true" Value="@_internalGridContext">
+    @{
+        StartCollectingColumns();
+    }
+    @if (!_manualGrid)
+    {
+        @ChildContent
+    }
+    <Defer>
+        @{
+            FinishCollectingColumns();
+        }
+        <ColumnsCollectedNotifier TGridItem="TGridItem" />
+
+        <table @ref=_gridReference
+               id="@Id"
+               class="@GridClass()"
```

Comment:
Why `class` call a method and  `style` call a property? This is more coherent to use the same "concept"